src/coreclr/jit/fgbasic.cpp

```diff
@@ -1097,6 +1104,11 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr, IL_OFFSET codeSize, Fixed
                     impResolveToken(codeAddr, &resolvedToken, CORINFO_TOKENKIND_Method);
                     methodHnd   = resolvedToken.hMethod;
                     isIntrinsic = eeIsIntrinsic(methodHnd);
+
+                    if (info.compCompHnd->isPinvoke(methodHnd))
```

Comment:
You may want to apply the same fix for calli. It has the same set of issues as direct PInvoke.